File: library/src/main/java/zhan/library/widget/HeaderLinearLayout.java
package zhan.library.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;
import zhan.library.delegate.ScrollHeaderDelegate;

/**
 * Created by hrz on 2016/7/8.
 */
public class HeaderLinearLayout extends LinearLayout {

    private ScrollHeaderDelegate mScrollHeaderDelegate;

    public HeaderLinearLayout(Context context) {
        super(context);
        init();
    }

    public HeaderLinearLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public HeaderLinearLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        mScrollHeaderDelegate = new ScrollHeaderDelegate(this);
    }

    @Override
    public void computeScroll() {
        mScrollHeaderDelegate.computeScroll();
    }

    public void setScrollShow(boolean isScrollShow) {
        mScrollHeaderDelegate.setScrollShow(isScrollShow);
    }

    public boolean isScrollShow() {
        return mScrollHeaderDelegate.isScrollShow();
    }

    public void scrollShow() {
        mScrollHeaderDelegate.scrollShow();
    }

    public void setDuration(int duration) {
        mScrollHeaderDelegate.setDuration(duration);
    }
}

File: double-pull-delegate/src/main/java/zhan/pull/widget/HeaderLinearLayout.kt
package zhan.pull.widget

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import zhan.pull.delegate.ScrollHeaderDelegate


class HeaderLinearLayout : LinearLayout {

    private var mScrollHeaderDelegate: ScrollHeaderDelegate? = null

    var isScrollShow: Boolean
        get() = mScrollHeaderDelegate!!.isScrollShow
        set(isScrollShow) {
            mScrollHeaderDelegate!!.isScrollShow = isScrollShow
        }

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        mScrollHeaderDelegate = ScrollHeaderDelegate(this)
    }

    override fun computeScroll() {
        mScrollHeaderDelegate!!.computeScroll()
    }

    fun scrollShow() {
        mScrollHeaderDelegate!!.scrollShow()
    }

    fun setDuration(duration: Int) {
        mScrollHeaderDelegate!!.setDuration(duration)
    }
}
